admin/src/services/abstractWidget.js

```diff
@@ -2,14 +2,16 @@ import { ADMIN_ACTIONS_WS_URL, BASE_URL_BACKEND } from "../config";
 import { createApiGet, createApiPost } from "../utils";
 
 export class AbstractWidgetService {
-    constructor(apiPath, errorHandler) {
+    constructor(apiPath, errorHandler, listenWS) {
```

Comment:
I think that instead of listenWS we should instead provide an array of reloadDataHandlers and open the websocket if it's not empty. That way it's more of a declarative approach that's closer to react instead of calling setReloadDataHandler for every component that needs it.